e2e/verify/verify.go

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package verify
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/buger/jsonparser"
+	"github.com/sylabs/singularity/e2e/internal/e2e"
+)
+
+type ctx struct {
+	env e2e.TestEnv
+}
+
+const successURL = "library://sylabs/tests/verify_success:1.0.1"
+const corruptedURL = "library://sylabs/tests/verify_corrupted:1.0.1"
+
+func getNameJSON(keyNum string) []string {
+	return []string{"SignerKeys", keyNum, "Signer", "Name"}
+}
+
+func getFingerprintJSON(keyNum string) []string {
+	return []string{"SignerKeys", keyNum, "Signer", "Fingerprint"}
+}
+
+func getLocalJSON(keyNum string) []string {
+	return []string{"SignerKeys", keyNum, "Signer", "KeyLocal"}
+}
+
+func getKeyCheckJSON(keyNum string) []string {
+	return []string{"SignerKeys", keyNum, "Signer", "KeyCheck"}
+}
+
+func getDataCheckJSON(keyNum string) []string {
+	return []string{"SignerKeys", keyNum, "Signer", "DataCheck"}
+}
+
+func (c *ctx) singularityVerifyKeyNum(t *testing.T) {
+	corruptedImage := filepath.Join(c.env.TestDir, "verify_corrupted.sif")
+	successImage := filepath.Join(c.env.TestDir, "verify_success.sif")
```

Comment:
Can we add these to the `c *ctx` that we pass to these test functions instead of reconstructing it for each test?